### PR TITLE
Fix some warnings and a build failure

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: quinn-h3/src/connection.rs [hunk 1]
```diff
@@ -267,15 +267,9 @@ impl ConnectionInner {
...
-                let mut pending = x.take().unwrap();
-                match Pin::new(&mut pending).poll(cx) {
-                    Poll::Ready(y) => Some((i, y)),
-                    Poll::Pending => {
-                        std::mem::replace(x, Some(pending));
-                        None
-                    }
-                }
+            .filter_map(|(i, x)| match Pin::new(x.as_mut().unwrap()).poll(cx) {
+                Poll::Ready(y) => Some((i, y)),
```
Comment (Ralith) [inline quinn-h3/src/connection.rs:271]: The previous code left `x` as `None` on `Poll::Ready`. Should this be preserved? If not, should `self.pending_uni` still contain `Option`? Not familiar with this code in general, but that seems suspect.

Comment (djc) [reply quinn-h3/src/connection.rs]: Oops, you're right. Fixed that up.

**Comment by Ralith** [comment]
Lint issue is unrelated; opened https://github.com/djc/quinn/pull/794 to resolve it.

**Comment by djc** [comment]
I've fixed the new lint. Take a look?

**Pull request merged**